docs/strategy-customization.md

```diff
@@ -639,6 +639,157 @@ Stoploss values returned from `custom_stoploss` must specify a percentage relati
+    define informative indicators.
+
+    Example usage:
+
+        @informative('1h')
+        def populate_indicators_1h(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
+            dataframe['rsi'] = ta.RSI(dataframe, timeperiod=14)
+            return dataframe
+
+    :param timeframe: Informative timeframe. Must always be equal or higher than strategy timeframe.
+    :param asset: Informative asset, for example BTC, BTC/USDT, ETH/BTC. Do not specify to use
+    current pair.
+    :param fmt: Column format (str) or column formatter (callable(name, asset, timeframe)). When not
+    specified, defaults to:
+    * {base}_{column}_{timeframe} if asset is specified and quote currency does match stake
+    curerncy. 
+    * {base}_{quote}_{column}_{timeframe} if asset is specified and quote currency does not match 
+    stake curerncy. 
+    * {column}_{timeframe} if asset is not specified.
+    Format string supports these format variables:
+    * {asset} - full name of the asset, for example 'BTC/USDT'.
+    * {base} - base currency in lower case, for example 'eth'.
+    * {BASE} - same as {base}, except in upper case.
+    * {quote} - quote currency in lower case, for example 'usdt'.
+    * {QUOTE} - same as {quote}, except in upper case.
+    * {column} - name of dataframe column.
+    * {timeframe} - timeframe of informative dataframe.
+    :param ffill: ffill dataframe after merging informative pair.
+    """
+```
+
+In most common case it is possible to easily define informative pairs by using a decorator. All decorated `populate_indicators_*` methods run in isolation,
+not having access to data from other informative pairs, in the end all informative dataframes are merged and passed to main `populate_indicators()` method.
+When hyperopting, use of hyperoptable parameter `.value` attribute is not supported. Please use `.range` attribute. See [optimizing an indicator parameter](hyperopt.md#optimizing-an-indicator-parameter)
+for more information.
+
+??? Example "Fast and easy way to define informative pairs"
+
+    Most of the time we do not need power and flexibility offered by `merge_informative_pair()`, therefore we can use a decorator to quickly define informative pairs.
+
+    ``` python
+
+    from datetime import datetime
+    from freqtrade.persistence import Trade
+    from freqtrade.strategy import IStrategy, informative
+
+    class AwesomeStrategy(IStrategy):
+        
+        # This method is not required. 
+        # def informative_pairs(self): ...
+
+        # Define informative upper timeframe for each pair. Decorators can be stacked on same 
+        # method. Available in populate_indicators as 'rsi_30m' and 'rsi_1h'.
+        @informative('30m')
+        @informative('1h')
+        def populate_indicators_1h(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
+            dataframe['rsi'] = ta.RSI(dataframe, timeperiod=14)
+            return dataframe
+
```

Comment:
Interrestingly, this case seems to work ONLY if there's other cases there as well.

so a "very" simple strategy limited to this decorator will not work (never call the decorated function).

```
    @informative('4h')
    def populate_indicators_1h(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        dataframe['rsiiiiii'] = ta.RSI(dataframe, timeperiod=14)
        return dataframe

    def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        print(dataframe.columns)
```

Adding a 2nd parameter will hoewever call it and make it work.

```
    @informative('4h', 'BTC/USDT)
    def populate_indicators_1h(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        dataframe['rsiiiiii'] = ta.RSI(dataframe, timeperiod=14)
        return dataframe
```

I wonder how tests can pass (probably because they directly load the strategy - and there's a TON of mocking/patching going on there ... as I also copied the new test-strategy and ran it - and it's missing the "rsi_30m" and "rsi_1h" columns too (only modification was a `print(dataframe.columns)` in `populate_indicators`.

Reply:
I fixed that (and few other things) but... I do hate the fix :( Problem is that creating informatives for each current pair requires access to dataprovider, but that is only available after class is initialized. So i deferred initialization of informatives to a method that is called after dataprovider is attached. This, however, means that users creating strategy object in the notebook would have to call this initialization method manually. I do not think this is a good solution, but i am not sure what to do about it. Dataprovider could be attached to class itself like it used to be in the past. However there are multiple issues with that. IIRC hyperopt had issues with that approach and i had to switch to attaching object to class instance itself. Also strategy loader creates strategy instance immediately, not giving us access to strategy class before object creation. Do you think it is reasonable to change strategy loader to return strategy class instead of instance?

Reply:
your comment gave me something to think ... and i think thecurrent solution will fail / not work (not because of the additional method ...).

You rely on the fact that the strategy is fully initialized, creating the "additional pairs lists" at instantiation time.
This is correct for static pairlists / backtest ... but doesn't account for dynamic pairlists (which can change over time).

As such - i think this PR will work only for the first pairlist ... and will start to have "missing" pairs the moment new pairs will be add.
my test ran into this (or a similar scenario) by accident - but it really exposed a different problem, which is that the "final" pairs will not be known when the strategy starts up.
You should "probably" be able to run into this by starting the bot with an existing database (with open trade) - where the open trade-pair is no longer in the pairlist (at the point you call _initialize, that pair is not yet part of the pairlist).

i think to solve this, we'll have to keep the "pairs" field empty for `@informative('30')` cases without defined pair (which means it'll use all pairs in the current whitelist).
As part of `gather_informative_pairs` - we'll then get the existing keys (where the pair is known) - and expand the "pair is unknown" cases (might need to be stored separately) to the "current" pairlist when actually gathering the pairs.